src/ops/ping.rs

```diff
@@ -1,18 +1,21 @@
 //! Run a BuildLoop for `shell.nix`, watching for input file changes.
 //! Can be used together with `direnv`.
-use crate::ops::error::{ok, OpResult};
-use crate::rpc;
+use crate::internal_proto;
+use crate::ops::error::{ok, ExitError, OpResult};
 use crate::NixFile;
 use std::convert::TryFrom;
 
 /// See the documentation for lorri::cli::Command::Ping_ for details.
-pub fn main(nix_file: NixFile) -> OpResult {
-    // TODO: set up socket path, make it settable by the user
-    let address = crate::ops::get_paths()?.daemon_socket_address();
-    let shell_nix = rpc::ShellNix::try_from(&nix_file).unwrap();
+pub fn main(nix_file: NixFile, addr: Option<String>) -> OpResult {
+    let address = addr
+        .ok_or(Err(()))
```

Comment:
The chained `or_else` retains the existing error that can arise if if `get_paths` fails.

To be honest, this was what I got working to align the Option of a provided socket address (which I needed for the integration tests) to work with the Result produced by `get_paths`. 

It's possible that a `map_err` would produce a better result, but that would be the case across the board - alternatively, the `get_paths` Err _ought_ to be human readable, no?

So I guess:

0. Do you have a better solution for welding an Option and a Result together?
0. Shouldn't the `Err` values here be readable? If not, I think that's a separate Issue.
0. On reflection, this seems like the model for handling provided socket addresses in the first place, so it would be good to get it right here.

Reply:
My bad. The `Err(())` here is immediately dealt with on the next line. I was worried it could be "shown" to the user, containing no information. But that is not the case. I couldn't find a better way to express this as a chain, either. As big a fan of chaining as I am, the following strikes me as a more obvious way of doing the same thing:

```rust
    let address = match addr {
        Some(address) => address,
        None => crate::ops::get_paths()?.daemon_socket_address(),
    };
```

Reply:
Yeah, let’s prefer `match` statements whenever possible, they are a lot easier to read usually.

Reply:
Updating this to use match was much better.